### ISSUE TITLE
Keyboard focus is lost from the text editor box intermittantly while using JAWS/screen raeder. (Accessibility)

### ISSUE THREAD

**Issue by GS-Avaluka** [microsoft/vscode#75199]
Using:
- Latest version of VSCode.
- JAWS v2019
(Note: I've been noticing this issue for a while now, though, and it wasn't just in this current version of VSCode.)

Potential Applicable JAWS Settings:
- I have Forms Mode set to "manual" so that it will not automatically throw me into forms mode when I tab into a text field or use the arrow-keys to navigate to a form field.

Overall Issue:
I sometimes lose keyboard focus from the text editor upon hitting the 'enter' key to add a new line to my code. This happens sporadically and I have 0 vision, so I am unable to tell what exactly is happening when this happens.
However, upon getting back to the text editor, the new line has successfully been added to the code file. It just seems that the focus is being lost sometimes when this happens.
My only guess is that perhaps the addition of the new line is visually scrolling the text editor and some action is breaking focus for the screen reader.

Steps to Reproduce:
- Turn on JAWS 2019.
- Open a file of code in VSC.
- Use the arrow-keys to browse through the code a bit.
- Use the 'enter' key to add a new line to the code.
- Keyboard focus is lost/forms mode ends.
- Navigate back to the text editor and hit enter on the text editor to turn on forms mode. Note that the new line has been added properly.

What Should Happen:
- Keyboard focus should remain in the text field until 'Escape' key is hit.

**Comment by egamma** [comment]
@GS-Avaluka sorry about the issue, have you tried NVDA as well, pls see https://code.visualstudio.com/docs/editor/accessibility#_screen-readers

**Comment by GS-Avaluka** [comment]
@egamma, no. I do not use NVDA and not all screen reader users do. I'm not going to switch to a completely foreign screen reader. I have 0 clue on the keyboard shortcuts for it and would be next to useless at it.

NVDA is free, though, so someone on the VSCode team that is proficient in it may be able to test to see if there is the same issue.

At the very least, the bug is not preventing me from coding ... I just lose focus periodically, have to use the arrow-keys to go down one and then hit enter again to re-enter the text field.

**Comment by egamma** [comment]
@GS-Avaluka as we wrote in the accessibility doc since the VS Code project is OSS we only test with the OSS NVDA. JAWS is currently not tested and therefore fully supported, sorry.  
We will test this scenario under NVDA.

**Comment by GS-Avaluka** [comment]
@egamma, you can download a JAWS demo version for free off of their website. It will work for 60 minutes without a license before the computer has to be rebooted for it to work for another 60 ... which I know isn't ideal, but it's far from being completely inaccessible to a developer for testing purposes.

I can only hope the same issue happens with NVDA so it can maybe be fixed, but if not ... oh well. I've tried NVDA before and was just not impressed with it. I realize it may have improved since I last tried it, but since I have access to JAWS for free through my government I have no reason to use it. I suspect the overall percentage of people using JAWS does surpass the percentage using NVDA, simply due to market share.

**Comment by isidorn** [comment]
I can not reproducce the steps you provided with NVDA.
This is caused by the JAWS setting Forms Mode I believe.
As far as I know NVDA it should never "steal" or move focus

As for not being impressed with NVDA, it is open source and I suggest to file your suggestions on how they can improve their product here https://github.com/nvaccess/nvda
Not sure what is the process of filling issues with JAWS but I would follow their steps and complain to them about this one.

Sorry but I believe there is nothing vscode can do to fix this, thus I am closing this one.
